app/services/candidate_interface/validate_course_query_string_params.rb

```diff
@@ -0,0 +1,46 @@
+module CandidateInterface
+  class ValidateCourseQueryStringParams
+    attr_accessor :return_course
+
+    def initialize(provider_code:, course_code:)
+      @provider_code = provider_code
+      @course_code = course_code
+      @can_apply_on_apply = false
+      @course_on_find = false
+      @return_course = nil
+    end
+
+    def execute
```

Comment:
What would you think of giving this a meaningful name like `determine_whether_course_is_on_find_or_apply` or something in that vein?